distributed/shuffle/_worker_plugin.py

```diff
@@ -51,6 +57,12 @@ def heartbeat(self) -> dict[ShuffleId, Any]:
         }
 
     def fail(self, shuffle_id: ShuffleId, run_id: int, message: str) -> None:
+        stale_run_id = run_id
+        if shuffle_id in self._stale_run_ids:
+            stale_run_id = max(self._stale_run_ids[shuffle_id], stale_run_id)
+
+        self._stale_run_ids[shuffle_id] = stale_run_id
```

Comment:
cosmetic; slightly nicer IMHO

```suggestion
        stale_run_id = self._stale_run_ids.setdefault(shuffle_id, run_id)
        if stale_run_id < run_id:
            self._stale_run_ids[shuffle_id] = run_id
```